cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java

```diff
@@ -60,87 +67,140 @@ public static void createReflectConfig(Path nativeConfigPath, Package currentPac
+                        new TypeToken<Map<String, String[]>>() {
+                        }.getType());
+                if (!testFileMockedFunctionMapping.isEmpty()) {
+                    ReflectConfigClass originalTestFileRefConfClz;
+                    for (Map.Entry<String, String[]> testFileMockedFunctionMappingEntry :
+                            testFileMockedFunctionMapping.entrySet()) {
+                        String moduleNameForTestClz = testFileMockedFunctionMappingEntry.getKey().split("-")[0];
+                        if (!moduleNameForTestClz.equals(name)) {
+                            continue;
+                        }
+                        String testFile = testFileMockedFunctionMappingEntry.getKey().split("-")[1];
+                        String[] mockedFunctions = testFileMockedFunctionMappingEntry.getValue();
+                        originalTestFileRefConfClz = new ReflectConfigClass(getQualifiedClassName(org, moduleNameForTestClz,
+                                version, testFile));
+                        for (int i = 0; i < mockedFunctions.length; i++) {
+                            originalTestFileRefConfClz.addReflectConfigClassMethod(
+                                    new ReflectConfigClassMethod(mockedFunctions[i]));
+                            originalTestFileRefConfClz.setUnsafeAllocated(true);
+                            originalTestFileRefConfClz.setAllDeclaredFields(true);
+                            originalTestFileRefConfClz.setQueryAllDeclaredMethods(true);
+                        }
+                        classList.add(originalTestFileRefConfClz);
+                    }
+                }
             }
+        }
 
+        //Add classes corresponding to ballerina source files
+        ReflectConfigClass originalBalFileRefConfClz;
+        Map<String, List<String>> mockFunctionClassMapping = new HashMap<>();
+        extractMockFunctionClassMapping(testSuiteMap, mockFunctionClassMapping);
+        for (Map.Entry<String, List<String>> mockFunctionClassMapEntry : mockFunctionClassMapping.entrySet()) {
+            String mockFunctionClass = mockFunctionClassMapEntry.getKey();
+            originalBalFileRefConfClz = new ReflectConfigClass(mockFunctionClass);
+            for (String originalMockFunction : mockFunctionClassMapEntry.getValue()) {
+                originalBalFileRefConfClz.addReflectConfigClassMethod(
+                        new ReflectConfigClassMethod(originalMockFunction));
+            }
+            originalBalFileRefConfClz.setQueryAllDeclaredMethods(true);
+            originalBalFileRefConfClz.setQueryAllDeclaredMethods(true);
+            originalBalFileRefConfClz.setUnsafeAllocated(true);
+            classList.add(originalBalFileRefConfClz);
         }
 
+        //Add test suite class
         ReflectConfigClass runtimeEntityTestSuiteRefConfClz = new ReflectConfigClass(
                 "org.ballerinalang.test.runtime.entity" + ".TestSuite");
```

Comment:
```suggestion
                "org.ballerinalang.test.runtime.entity.TestSuite");
```
Is there some reason we can't have it like this?